services/search/pkg/search/service.go

```diff
@@ -229,7 +231,7 @@ func (s *Service) searchIndex(ctx context.Context, req *searchsvc.SearchRequest,
 		rootName         string
 		permissions      *provider.ResourcePermissions
 	)
-	mountpointPrefix := ""
+	mountpointPrefix := req.GetRef().GetPath()
```

Comment:
When scoping the search to a specific location we can also adapt the call to `ListStorageSpaces` to only fetch the single space that's relevant, no?

Reply:
@aduffeck If I add the Opaque to the ListStorageSpacesRequest, should it work? It doesn't work for me the ListStorageSpaces returns the empty list.
`	listSpacesRes, err := gatewayClient.ListStorageSpaces(ctx, &provider.ListStorageSpacesRequest{
		Opaque: utils.AppendPlainToOpaque(nil, "path", ref.GetPath()),
		Filters: []*provider.ListStorageSpacesRequest_Filter{
			{
				Type: provider.ListStorageSpacesRequest_Filter_TYPE_USER,
				Term: &provider.ListStorageSpacesRequest_Filter_User{User: currentUser.GetId()},
			},
			{
				Type: provider.ListStorageSpacesRequest_Filter_TYPE_SPACE_TYPE,
				Term: &provider.ListStorageSpacesRequest_Filter_SpaceType{SpaceType: "+grant"},
			},
		},
	})`

Reply:
@aduffeck  adde the filter [https://github.com/owncloud/ocis/pull/6663/files#diff-1829b55dcbf1442832a7a05f86467898b1649e7096eb930b707663babee423b8R93](url)